server/dns_server.mli

```diff
@@ -76,6 +76,13 @@ val handle_question : t -> Packet.Question.t ->
 (** [handle_question t question] handles the DNS query [question] by looking
     it up in the trie of [t]. *)
 
+val update_data : Dns_trie.t -> 'a Domain_name.t ->
+  Dns.Packet.Update.prereq list Domain_name.Map.t
+  * Dns.Packet.Update.update list Domain_name.Map.t ->
+  ( Dns_trie.t * (Domain_name.Set.elt * Dns.Soa.t) list,
+   Dns.Rcode.t )
+  result
```

Comment:
there's need for a docstring describing the semantics -- and what will be done, what won't be (e.g. authentication).

also, the return value should be described in detail -- what the second projection is supposed to contain, etc.

I also wonder what the operation for a DNS server should be -- so once an update_data has been performed, there may be need to notify secondary servers -- how's this supposed to happen? and with your cap'n'proto work, how do you notify the other servers when a DNS update is performed?